include/irr/builtin/material_compiler/glsl/common_declarations.glsl

```diff
@@ -0,0 +1,28 @@
+#ifndef _IRR_BUILTIN_MATERIAL_COMPILER_GLSL_COMMON_DECLARATIONS_INCLUDED_
+#define _IRR_BUILTIN_MATERIAL_COMPILER_GLSL_COMMON_DECLARATIONS_INCLUDED_
+
+#include <irr/builtin/glsl/virtual_texturing/extensions.glsl>
+
+#define instr_t uvec2
+#define reg_t uint
+#define params_t mat4x3
+#define bxdf_eval_t vec3
+#define eval_and_pdf_t vec4
+
+#define INSTR_1ST_DWORD(instr) (instr).x
+#define INSTR_2ND_DWORD(instr) (instr).y
+
+struct bsdf_data_t
+{
+	uvec4 data[sizeof_bsdf_data];
+};
+
+struct instr_stream_t
+{
+	uint offset;
+	uint count;
+};
+
+#define CIE_XYZ_Luma_Y_coeffs vec3(0.2126,0.7152,0.0722)
```

Comment:
you can get these from the OETF and colorspace headers (just transpose the matrix first, so you can extract the [1] row)